### PR TITLE
Updating APM UI Page to APM Concepts & Walkthrough

### PR DESCRIPTION
**Pull request by andrewsouthard1**
### What does this PR do?

This PR updates the APM UI landing page to an APM Concepts & Walkthrough. 

### Motivation

User onboarding - see internal request here: https://cl.ly/f1117692e87c This PR also clarifies the term tags/metadata into the single term -> tags. This is consistent with how we have it in our tracers and also how it will now be displayed in-product. https://cl.ly/09c2f1ca6367

### Preview link

https://docs-staging.datadoghq.com/andrew/apm-concepts/tracing/visualization/
https://docs-staging.datadoghq.com/andrew/apm-concepts/tracing/advanced/adding_metadata_to_spans/

### Additional Notes

Would like this merged by EOD Friday - iterating on feedback today.

### CODE REVIEW

File: content/en/tracing/visualization/_index.md [hunk 1]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
 ## Services
 
-**A service is a set of processes that do the same job.**
-For instance, a simple web application may consist of two services:
+After [instrumenting your application][1], the [Services List][2] will be your main landing page to get started with APM data. 
+
+{{< img src="tracing/visualization/service_list.png" alt="service list" responsive="true">}}
 
-* A single `webapp` service and a single `database` service.
+Services are the building blocks of modern microservice architectures - broadly a service groups together endpoints for the purposes of scaling instances. For example, a group of URL endpoints may be grouped together under an API service. Another example may be a group of DB queries that are grouped together within one database service. Yet another example may be a group of periodic jobs configured in the crond service. In the screenshot below - in this microservice distributed system for an e-commerce site builder, we have a `web-store`, `ad-server`, `payment-db`, `auth-service` all represented as services in APM.
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
Services are the building blocks of modern microservice architectures - broadly, a service groups together endpoints for the purposes of scaling instances. For example, a group of URL endpoints may be grouped together under an API service. Another example may be a group of DB queries that are grouped together within one database service. Yet another example may be a group of periodic jobs configured in the crond service. In the screenshot below - in this microservice distributed system for an e-commerce site builder, we have a `web-store`, `ad-server`, `payment-db`, `auth-service` all represented as services in APM.
```

File: content/en/tracing/visualization/_index.md [hunk 2]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
 
-While a more complex environment may break it out into 6 services:
+{{< img src="tracing/visualization/service_map.png" alt="service map" responsive="true">}} 
 
-* 3 separate services: `webapp`, `admin`, and `query`.
-* 3 separate external service:  `master-db`,  `replica-db`, and `yelp-api`.
+All services can be found in the [Service List][2] and visually represented on the [Service Map][3]. Each Service has its own [Service Page][4] where [trace metrics](#trace-metrics) like throughput, latency and error rates can be viewed and inspected. You can use these metrics for the service and create dashboard widgets, create monitors and see the performance of every resource such as a web endpoint or database query belonging to the service.
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
All services can be found in the [Service List][2] and visually represented on the [Service Map][3]. Each Service has its own [Service Page][4] where [trace metrics](#trace-metrics) like throughput, latency, and error rates can be viewed and inspected. You can use these metrics for the service and create dashboard widgets, create monitors, and see the performance of every resource such as a web endpoint or database query belonging to the service.
```

File: content/en/tracing/visualization/_index.md [hunk 3]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
 
-APM automatically assigns names to your services; however you can also name them explicitly. See instructions for: [Go][1], [Java][2], [Python][3], [Ruby][4].
+{{< img src="tracing/visualization/service_page.gif" alt="service page" responsive="true">}} 
 
-Service names:
+<div class="alert alert-info">
+Don’t see the http endpoints you were expecting on the Service Page? In APM endpoints are connected to a service by more than the service name - it is also done via the `span.name` of the entry-point span of the trace. For example, on the web-store service above we’re viewing endpoints where `web.request` is the entry-point span. More info on this <a href="/tracing/faq/resource-trace-doesn-t-show-up-under-correct-service/"> here.
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
Don’t see the http endpoints you were expecting on the Service Page? In APM, endpoints are connected to a service by more than the service name - it is also done using the `span.name` of the entry-point span of the trace. For example, on the web-store service above we’re viewing endpoints where `web.request` is the entry-point span. For more information, see  <a href="/tracing/faq/resource-trace-doesn-t-show-up-under-correct-service/">Resource/Trace doesn't show up under correct service</a>.
```

File: content/en/tracing/visualization/_index.md [hunk 4]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
+## Resources
 
-**Note**: Services must have a type attached, APM automatically assigns services one of four types: web, database, cache, custom.
+In APM, a Service corresponds to a group of Resources that serve a particular domain of a customer application. Resources could typically be an instrumented web endpoint, database query, or background job. For a web service, these resources can be dynamic web endpoints that are grouped by a static span name -  `web.request`. In a database service, these would be database queries with the span name `db.query`. For example the `web-store` service has automatically instrumented resources - web endpoints - which handle checkouts, updating_carts, add_item, etc. Each resource has its own [Resource Page][5] with [trace metrics](#trace-metrics) scoped to the specific endpoint. Trace metrics can be used like any other Datadog metric - they are exportable to a dashboard or can be used to create monitors. The Resource Page will also show the span summary widget with an aggregate view of [spans](#spans) for all [traces](#trace), latency distribution of requests, and then traces which show requests made to this endpoint.
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
In APM, a Service corresponds to a group of Resources that serve a particular domain of a customer application. Resources could typically be an instrumented web endpoint, database query, or background job. For a web service, these resources can be dynamic web endpoints that are grouped by a static span name - `web.request`. In a database service, these would be database queries with the span name `db.query`. For example, the `web-store` service has automatically instrumented resources - web endpoints - which handle checkouts, updating_carts, add_item, etc. Each resource has its own [Resource Page][5] with [trace metrics](#trace-metrics) scoped to the specific endpoint. Trace metrics can be used like any other Datadog metric - they are exportable to a dashboard or can be used to create monitors. The Resource Page will also show the span summary widget with an aggregate view of [spans](#spans) for all [traces](#trace), latency distribution of requests, and then traces which show requests made to this endpoint.
```

File: content/en/tracing/visualization/_index.md [hunk 5]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
+
+{{< img src="tracing/visualization/trace_metric_dashboard.gif" alt="trace metrics dashboard" responsive="true">}} 
+
+{{% /tab %}}
+
+{{% tab "Monitoring" %}}
+
+### Monitoring
+
+Trace metrics can be used in a monitor like any other datadog metric - APM monitors can be set up on the [Service Page][4] or [Resource Page][5] with a set of suggested monitors to get started, or on the [New Monitors page][18].
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
Trace metrics can be used in a monitor - APM monitors can be set up on the [Service Page][4] or [Resource Page][5] with a set of suggested monitors to get started, or on the [New Monitors page][18].
```

File: content/en/tracing/visualization/_index.md [hunk 6]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
-* **Cannot exceed 5000 bytes**
+{{< img src="tracing/visualization/analyzed_span.gif" alt="analyzed span" responsive="true">}} 
 
-[Alert][12] on any resource level metric. Read more about monitoring resources in APM on the [resource dashboard][13] page.
 
-### Trace
+## Tagging Spans
 
-**A trace is used to track the time spent by an application processing a single operation, each trace consists of one or more spans.**
+You can tag spans in the form of key-value pairs to isolate in the Trace View or slice/dice on in Trace Search & Analytics. Tags can be either added to a single span or globally to all spans. On this request `merchant.store_name`, `merchant.tier`, etc. have been added as tags to the span.
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
You can tag spans in the form of key-value pairs to isolate in the Trace View or to use to filter in Trace Search & Analytics. Tags can be either added to a single span or globally to all spans. On this request `merchant.store_name`, `merchant.tier`, etc. have been added as tags to the span.
```

File: content/en/tracing/visualization/_index.md [hunk 7]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
 
-For example, a trace can be used to track the entire time spent processing a complicated web request. Even though the request may require multiple resources and machines to handle the request, all of these function calls and sub-requests would be encapsulated within a single trace.
+{{< img src="tracing/visualization/span_tag.png" alt="span tag" responsive="true">}} 
 
-### Spans
+To get started tagging spans in your application, our walkthrough is [here][15].
```
Comment (apigirl) [inline content/en/tracing/visualization/_index.md]:
```suggestion
To get started tagging spans in your application, check out our [walkthrough][15].
```

File: content/en/tracing/visualization/_index.md [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Visualizing your APM data
+title: APM Concepts & Walkthrough
```
[the remark this answers is not in the document]
Comment (andrewsouthard1) [reply content/en/tracing/visualization/_index.md]: I like the idea of a walkthrough as it seems less threatening that a glossary.

File: content/en/tracing/visualization/_index.md (hunk 4 above)
Comment (andrewsouthard1) [reply content/en/tracing/visualization/_index.md]: Reworded this one.

File: content/en/tracing/visualization/_index.md (hunk 3 above)
Comment (andrewsouthard1) [reply content/en/tracing/visualization/_index.md]: Adjusted this one as well with a comma

File: content/en/tracing/visualization/_index.md [hunk 9]
```diff
@@ -22,79 +22,134 @@ further_reading:
...
+
+| Concept               | Description             |
+|-----------------------|-------------------------|
+| Service               | Services are the building blocks of modern microservice architectures - broadly a service groups together endpoints for the purposes of scaling instances.                                                            |
+| Resource              | In APM, a Service corresponds to a group of Resources that serve a particular domain of a customer application. Resources could typically be an instrumented web endpoint, database query, or background job   |
+| Trace                 | A trace is used to track the time spent by an application processing a single request and the status of this request, each trace consists of one or more spans.                                                          |
+| Span                  | Datadog APM will automatically collect trace metrics that can be used to identify and alert on hits, errors or latency and are tagged by the host receiving traces along with the service or resource.                                                                                                   |
```
[the remark this answers is not in the document]
Comment (andrewsouthard1) [reply content/en/tracing/visualization/_index.md]: Reworded this one to remove "Datadog APM"

File: content/en/tracing/visualization/_index.md [hunk 10]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
+
+Trace metrics can be used in a monitor - APM monitors can be set up on the [Service Page][4] or [Resource Page][5] with a set of suggested monitors to get started, or on the [New Monitors page][18].
+
+{{< img src="tracing/visualization/trace_metric_monitor.gif" alt="trace metrics monitor" responsive="true">}} 
+
+
+## Trace Search & Analytics
 
-**A Resource is a particular action for a service**.
+Trace Search & Analytics is used to filter [APM Events](#apm-event) by user-defined tags such as `customer_id`, `error_type`, or `app_name` or infrastructure tags to help troubleshoot and filter your requests. It allows deep exploration of the web requests flowing through your service along with being able to search, graph, and monitor on 100% throughput of hits, errors, and latency. This feature can be enabled with [automatic configuration][14].
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
Trace Search & Analytics is used to filter [APM Events](#apm-event) by user-defined tags (customer_id, error_type, app_name, etc.) or infrastructure tags. This allows deep exploration of the web requests flowing through your service along with being able to search, graph, and monitor on 100% throughput of hits, errors, and latency. This feature can be enabled with [automatic configuration][14].
```

File: content/en/tracing/visualization/_index.md [hunk 11]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-* **For a web application**: some examples might be a canonical URL, such as `/user/home` or a handler function like `web.user.home` (often referred to as "routes" in MVC frameworks).
-* **For a SQL database**: a resource is the query itself, such as `SELECT * FROM users WHERE id = ?`.
+{{< vimeo 278748711 >}}
 
-Resources should be grouped together under a canonical name, like `/user/home` rather than have `/user/home?id=100` and `/user/home?id=200` as separate resources. APM automatically assigns names to your resources; however you can also name them explicitly. See instructions for: [Go][9], [Java][2], [Python][10], [Ruby][11].
+## APM Event
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
## APM event
```

File: content/en/tracing/visualization/_index.md [hunk 12]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-These resources can be found after clicking on a particular [service][8].
+APM Events represent 100% throughput of a request and can be used to search, query, and monitor in Trace Search & Analytics by the [tags](#span-tags) included on the span. After enabling Trace Search & Analytics, the tracing client will analyze an entry-point span for web services by default, with the ability to [configure additional services][19] in your application. For example, a java service with 100 requests will generate 100 APM Events from its `servlet.request` spans. In this example `web-store` service, after setting `DD_TRACE_ANALYTICS_ENABLED=true` now all `rack.request` spans will be analyzed and are available to use in Trace Search & Analytics. For this example, you can graph the top 10 merchants highest latency in the 99th percentile. `merchant_name` is a user defined tag that was applied to the span in the application.
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
APM events represent 100% throughput of a request and can be used to search, query, and monitor in Trace Search & Analytics by the [tags](#span-tags) included on the span. After enabling Trace Search & Analytics, the tracing client analyzes an entry-point span for web services by default, with the ability to [configure additional services][19] in your application. For example, a Java service with 100 requests generates 100 APM events from its `servlet.request` spans. If you set `DD_TRACE_ANALYTICS_ENABLED=true` the `web-store` service analyzes all `rack.request` spans and makes them available in Trace Search & Analytics. For this example, you can graph the top 10 merchants highest latency in the 99th percentile. `merchant_name` is a user defined tag that was applied to the span in the application.
```

File: content/en/tracing/visualization/_index.md [hunk 13]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-Resource names:
+{{< img src="tracing/visualization/analyzed_span.gif" alt="analyzed span" responsive="true">}} 
 
-* **Must be lowercase, alphanumeric characters**
-* **Cannot exceed 5000 bytes**
 
-[Alert][12] on any resource level metric. Read more about monitoring resources in APM on the [resource dashboard][13] page.
+<div class="alert alert-info">
+You can run an estimate on the number of APM Events that would be generated from your services <a href="https://app.datadoghq.com/apm/docs/trace-search?env=datad0g.com"> here </a>. After ingestion, you can also filter APM Events from 100% to a lower percentage on a service-by-service level, reducing billable APM Events
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
You can run an estimate on the number of APM events that would be generated from your services with the <a href="https://app.datadoghq.com/apm/docs/trace-search">Analyzed Span Estimator</a>. After ingestion, you can filter APM events from 100% to a lower percentage on a service-by-service level under <a href="https://app.datadoghq.com/apm/settings">APM settings</a>. This reduces billable APM events.
```

File: content/en/tracing/visualization/_index.md [hunk 14]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
+<a href="https://app.datadoghq.com/apm/settings?env=datad0g.com"> here </a>.
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
```

File: content/en/tracing/visualization/_index.md [hunk 15]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
-* **Cannot exceed 5000 bytes**
 
-[Alert][12] on any resource level metric. Read more about monitoring resources in APM on the [resource dashboard][13] page.
+<div class="alert alert-info">
+You can run an estimate on the number of APM Events that would be generated from your services <a href="https://app.datadoghq.com/apm/docs/trace-search?env=datad0g.com"> here </a>. After ingestion, you can also filter APM Events from 100% to a lower percentage on a service-by-service level, reducing billable APM Events
+<a href="https://app.datadoghq.com/apm/settings?env=datad0g.com"> here </a>.  
+</div>
 
-### Trace
+## Span Tags
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
## Span tags
```

File: content/en/tracing/visualization/_index.md [hunk 16]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-**A trace is used to track the time spent by an application processing a single operation, each trace consists of one or more spans.**
+You can tag spans in the form of key-value pairs to correlate a request in the Trace View or to use to filter in Trace Search & Analytics. Tags can be either added to a single span or globally to all spans. On this request `merchant.store_name`, `merchant.tier`, etc. have been added as tags to the span.
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
Tag spans in the form of key-value pairs to correlate a request in the *Trace View* or filter in *Trace Search & Analytics*. Tags can be added to a single span or globally to all spans. For the example below, the requests (`merchant.store_name`, `merchant.tier`, etc.) have been added as tags to the span.
```

File: content/en/tracing/visualization/_index.md [hunk 17]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-For example, a trace can be used to track the entire time spent processing a complicated web request. Even though the request may require multiple resources and machines to handle the request, all of these function calls and sub-requests would be encapsulated within a single trace.
+{{< img src="tracing/visualization/span_tag.png" alt="span tag" responsive="true">}} 
 
-### Spans
+To get started tagging spans in your application, check out our [walkthrough][15].
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
To get started tagging spans in your application, check out this [walkthrough][15].
```

File: content/en/tracing/visualization/_index.md [hunk 18]
```diff
@@ -21,80 +21,127 @@ further_reading:
...
 
-**A span represents a logical unit of work in the system.**
+After a tag has been added to a span, to search and query on the tag in Trace Search & Analytics, click on it and add it as a [facet][16]. Once this is done, the value of this tag is stored for all new traces and can be used in the search bar, the Facet Panel, and in the Trace graph query.
```
Comment (ruthnaebeck) [inline content/en/tracing/visualization/_index.md]:
```suggestion
After a tag has been added to a span, search and query on the tag in Trace Search & Analytics by clicking on the tag to add it as a [facet][16]. Once this is done, the value of this tag is stored for all new traces and can be used in the search bar, facet panel, and trace graph query.
```

File: content/en/tracing/advanced/adding_metadata_to_spans.md [hunk 19]
```diff
@@ -1,6 +1,8 @@
 ---
-title: Adding Metadata to Spans
+title: Adding Span Tags
 kind: documentation
+aliases:
+  - /tracing/advanced/adding_metadata_to_spans/
```
[the remark this answers is not in the document]
Comment (andrewsouthard1) [reply content/en/tracing/advanced/adding_metadata_to_spans.md]: Removed in last commit

**Pull request merged**